qiskit_nature/problems/sampling/protein_folding/protein_folding_problem.py

```diff
@@ -0,0 +1,85 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Defines a protein folding problem that can be passed to algorithms."""
+from typing import Union, List
+
+from qiskit.opflow import PauliSumOp, PauliOp
+
+from .peptide.peptide import Peptide
+from .interactions.interaction import Interaction
+from .penalty_parameters import PenaltyParameters
+from .qubit_op_builder import QubitOpBuilder
+from .qubit_utils import qubit_number_reducer
+from ..sampling_problem import SamplingProblem
+
+
+class ProteinFoldingProblem(SamplingProblem):
+    """Defines a protein folding problem that can be passed to algorithms."""
+
+    def __init__(
+        self, peptide: Peptide, interaction: Interaction, penalty_parameters: PenaltyParameters
+    ):
+        """
+        Args:
+            peptide: A peptide object that defines the protein subject to the folding problem.
+            interaction: A type of interaction between the beads of the peptide.
+            penalty_parameters: Parameters that define the strength of constraints enforcing in
+                                the problem.
+        """
+        self._peptide = peptide
+        self._interaction = interaction
+        self._penalty_parameters = penalty_parameters
+        self._pair_energies = interaction.calc_energy_matrix(
+            len(peptide.get_main_chain), peptide.get_main_chain.main_chain_residue_sequence
+        )
+        self._qubit_op_builder = QubitOpBuilder(
+            self._peptide, self._pair_energies, self._penalty_parameters
+        )
+        self._unused_qubits: List[int] = []
+
+    def qubit_op(self) -> Union[PauliSumOp, PauliOp]:
```

Comment:
@pbark What do you think? I would be ok with dropping full.

Reply:
I would keep also the full in case we want to extend functionalities in the future. For example one may want to keep the full list in order to encode (or grow) side chains from this direction. Possibly you can make the qubit_op_full private so that is not exposed in the documentation and the end user and keep it for future reference.

Reply:
Yep, that's what I did - private. Thanks.